src/components/ProgramForm/ThreeColProgram/List/LessonList/utils.ts

```diff
@@ -0,0 +1,149 @@
+import { moveItemOnTree } from '@atlaskit/tree';
+import type {
+  TreeItem,
+  TreeSourcePosition,
+  TreeDestinationPosition,
+  TreeData,
+  ItemId,
+} from '@atlaskit/tree';
+import type { Dispatch, SetStateAction } from 'react';
+
+// UTIL TYPES
+type StringifyId<Obj extends { id?: number }> = Omit<Obj, 'id'> & { id: string };
+type LessonDeeplyStringifyId = Omit<StringifyId<API.Lesson>, 'lessons' | 'topics'> & {
+  lessons?: StringifyId<API.Lesson>[];
+  topics?: StringifyId<API.Topic>[];
+};
+
+const mutateTopicsId = (topics: API.Topic[]): StringifyId<API.Topic>[] =>
+  topics.map((t) => ({ ...t, id: `topic-${t.id}` }));
+const mutateLessonsId = (lessons: API.Lesson[]): StringifyId<API.Lesson>[] =>
+  lessons.map((l) => ({ ...l, id: l.isNew ? `new-${l.id}` : `lesson-${l.id}` }));
```

Comment:
why `mutate` ?

Reply:
Id could be the same and we wouldn't know if it's lesson or topic. Name could be probably better btw :D